docs/howto/pipeline_testing.md

```diff
@@ -0,0 +1,183 @@
+# HOWTO: Writing pipeline tests for a package
+
+## Introduction
+
+Elastic Packages are comprised of data streams. A pipeline test exercises ingest pipelines defined for a package's data stream.
+
+## Conceptual process
+
+Conceptually, running a pipeline test involves the following steps:
+
+1. Deploy the Elasticsearch instance (part of the Elastic Stack). This step takes time so it should typically be done once as a pre-requisite to running pipeline tests on multiple data streams.
+1. Upload ingest pipelines to be tested.
+1. Use [Simulate API](https://www.elastic.co/guide/en/elasticsearch/reference/master/simulate-pipeline-api.html) to process logs/metrics with the ingest pipeline.
+1. Compare generated results with expected ones.
+
+## Limitations
+
+At the moment pipeline tests have limitations. The main ones are:
+* Results of simulation can be different from real documents. They depend on the input data which may not have all fields defined.
```

Comment:
I didn't quite understand this one. Can you explain to me with a toy example?

Reply:
It's probably bad wording.

As you're only testing the ingest pipeline, you can prepare mocked documents with unreal fields, different from ones collected in Beats. Also the other way round, you can skip most of fields and as examples use tiny documents with minimal set of fields just to satisfy the pipeline validation.

Reply:
Thanks for clarifying. Maybe just use the wording you gave me in the documentation? Except in place of "unreal" I would say "imaginary" or "fictitious".

Reply:
Fixed!